qiskit_ibm_provider/transpiler/passes/scheduling/pad_delay.py

```diff
@@ -72,6 +72,4 @@ def _pad(
             return
 
         time_interval = t_end - t_start
-        self._apply_scheduled_op(
-            block_idx, t_start, Delay(time_interval, self._dag.unit), qubit
-        )
+        self._apply_scheduled_op(block_idx, t_start, Delay(time_interval, "dt"), qubit)
```

Comment:
Is it ok to change the unit like this?  If it's required to be `dt`, should something throw an error if it isn't?

Reply:
Updated the padder to ensure this is the case. The current problem is that time_units are not propagated to control-flow child blocks.

Reply:
Added a check here 8f2ec6a